source/numerics/vector_tools_boundary.inst.in

```diff
@@ -220,22 +220,38 @@ for (deal_II_dimension : DIMENSIONS; deal_II_space_dimension : SPACE_DIMENSIONS)
         AffineConstraints<double> &,
         const hp::MappingCollection<deal_II_dimension> &);
       template void
-      project_boundary_values_curl_conforming_l2<deal_II_dimension>(
+      project_boundary_values_curl_conforming_l2(
         const DoFHandler<deal_II_dimension> &,
         const unsigned int,
-        const Function<deal_II_dimension> &,
+        const Function<deal_II_dimension, double> &,
         const types::boundary_id,
         AffineConstraints<double> &,
         const Mapping<deal_II_dimension> &);
       template void
-      project_boundary_values_curl_conforming_l2<deal_II_dimension>(
+      project_boundary_values_curl_conforming_l2(
+        const DoFHandler<deal_II_dimension> &,
+        const unsigned int,
+        const Function<deal_II_dimension, std::complex<double>> &,
+        const types::boundary_id,
+        AffineConstraints<std::complex<double>> &,
+        const Mapping<deal_II_dimension> &);
```

Comment:
Would you mind guarding this instantiation and the one below by
```
#ifdef DEAL_II_WITH_COMPLEX_VALUES
```
and the tests by `with_complex_values=on`?

Reply:
I added the guards, the changelog entry and `with_complex_values=on`